### PR TITLE
Add support of \dn [PATTERN] command

### PR DESCRIPTION
**Pull request by CyberDem0n**
This command is used in standard psql for listing of schemas

### CODE REVIEW

File: pgcli/packages/pgspecial.py [hunk 1]
```diff
@@ -25,6 +25,26 @@ def parse_special_command(sql):
...
+def list_schemas(cur, pattern, verbose):
+    """
+    Returns (rows, headers, status)
+    """
+    sql = """SELECT n.nspname AS "Name",
+    pg_catalog.pg_get_userbyid(n.nspowner) AS "Owner"
+    FROM pg_catalog.pg_namespace n WHERE n.nspname """
+
+    if pattern:
+        sql += "~ '^(" + pattern.replace('*', '.*') + ")$'"
```
Comment (amjith) [inline pgcli/packages/pgspecial.py]: You might have to use [sql_name_pattern](https://github.com/amjith/pgcli/blob/master/pgcli/packages/pgspecial.py#L656) to do this.

**Comment by amjith** [comment]
Apart from the pattern conversion stuff, things look good. If you can update that, I'll test it locally and merge it in. 

Thanks for doing this.

File: pgcli/packages/pgspecial.py (hunk 1 above)
Comment (CyberDem0n) [inline pgcli/packages/pgspecial.py]: sql_name_pattern is not really doing what we need.

First of all \dn accepts only schema name, so we don't need to extract schema-table pair from it.
Then, any object may contain "." in it's name. For example: "my.schema"."my.table". Moreover, object name could also contain even " (double quote) character.
I think we should reimplement sql_name_pattern by using the same approach as in original processSQLNamePattern function: http://git.postgresql.org/gitweb/?p=postgresql.git;a=blob;f=src/bin/pg_dump/dumputils.c;h=55509866ff5c692288e8b31074a6912020b381ca;hb=master#l956

Comment (amjith) [inline pgcli/packages/pgspecial.py]: I did base the sql_name_pattern() on the original from psql, but I guess I didn't fully cover all the cases and chose to go with a simplified impl. 

Would you like to take a stab at it or should I file an issue for it and get back to it at a later time?

Comment (CyberDem0n) [inline pgcli/packages/pgspecial.py]: Ok, I will take it over.

Comment (CyberDem0n) [inline pgcli/packages/pgspecial.py]: Done with sql_name_patter and related stuff. Please check.

**Comment by amjith** [comment]
:+1: Thank you!!

Looks good. :soccer:

**Pull request merged**